pytorch_lightning/trainer/trainer.py

```diff
@@ -764,6 +781,62 @@ def test(
             results = self.__test_using_best_weights(ckpt_path, test_dataloaders)
 
         self.teardown('test')
+        self.running_stage = RunningStage.UNDEFINED
+
+        return results
+
+    def predict(
+        self,
+        model: Optional[LightningModule] = None,
+        test_dataloaders: Optional[Union[DataLoader, List[DataLoader]]] = None,
```

Comment:
Might be good to select a better name than test data loader, maybe even just dataloaders

Reply:
Done !